graylog2-web-interface/src/components/sidecars/failure-tracking/SidecarFailureTrackingList.test.tsx

```diff
@@ -0,0 +1,135 @@
+    active: false,
+    node_id: 'node_id',
+  }];
+  const collectors = [{
+    service_type: '',
+    node_operating_system: '',
+    name: '',
+    validation_parameters: '',
+    executable_path: '',
+    execute_parameters: '',
+    default_template: '',
+    id: '',
+  }];
+  const pagination = { total: 11, count: 10, page: 1, per_page: 25, query: '' };
+  const query = 'query';
+  const onlyActive = false;
+  const sort = { field: 'last_seen', order: 'desc' };
+  const pageSizes = [25, 50, 100];
+
+  const handlePageChange = jest.fn();
+  const handleQueryChange = jest.fn();
+  const handleSortChange = jest.fn();
+  const toggleShowInactive = jest.fn();
+
+  it('should render empty list', async () => {
+    const { findByText } = render(
+      <SidecarFailureTrackingList sidecars={[]}
+                                  collectors={[]}
+                                  pagination={pagination}
+                                  query={query}
+                                  onlyActive={onlyActive}
+                                  sort={sort}
+                                  pageSizes={pageSizes}
+                                  onPageChange={handlePageChange}
+                                  onQueryChange={handleQueryChange}
+                                  onSortChange={handleSortChange}
+                                  toggleShowInactive={toggleShowInactive} />,
+    );
+
+    await findByText(/There are no sidecars with failures/);
+  });
+
+  it('should render list with all features', async () => {
+    const { findByText, getByRole, getByTitle, getByLabelText } = render(
+      <SidecarFailureTrackingList sidecars={sidecars}
+                                  collectors={collectors}
+                                  pagination={pagination}
+                                  query={query}
+                                  onlyActive={onlyActive}
+                                  sort={sort}
+                                  pageSizes={pageSizes}
+                                  onPageChange={handlePageChange}
+                                  onQueryChange={handleQueryChange}
+                                  onSortChange={handleSortChange}
+                                  toggleShowInactive={toggleShowInactive} />,
+    );
+
+    await findByText('Sidecar');
+    await findByText('Collector');
+    await findByText('Last Seen');
+    await findByText('Status');
+    await findByText('Error Message');
+    await findByText('Verbose Message');
+
+    const findButton = getByRole('button', { name: 'Find' });
+    const resetButton = getByRole('button', { name: 'Reset' });
+    const hideActiveButton = getByRole('button', { name: 'Hide inactive sidecars' });
+    const sortByNameButton = getByTitle('node_name');
+    const sortByLastSeenButton = getByTitle('last_seen');
+    const pageSizeInput = getByLabelText('Show');
+
+    fireEvent.change(pageSizeInput, { target: { value: 100 } });
```

Comment:
I think I would create own tests for each handler, this way when a test fails it's easier to understand which functionality failed just by the test description.